src/simulation.cpp

```diff
@@ -464,17 +463,32 @@ void finalize_batch()
   }
 
   // Write out surface source if requested.
-  if (settings::surf_source_write &&
-      simulation::current_batch == settings::n_batches) {
-    auto filename = settings::path_output + "surface_source";
-    auto surf_work_index =
-      mpi::calculate_parallel_index_vector(simulation::surf_source_bank.size());
-    gsl::span<SourceSite> surfbankspan(simulation::surf_source_bank.begin(),
-      simulation::surf_source_bank.size());
-    if (settings::surf_mcpl_write) {
-      write_mcpl_source_point(filename.c_str(), surfbankspan, surf_work_index);
-    } else {
-      write_source_point(filename.c_str(), surfbankspan, surf_work_index);
+  if (settings::surf_source_write) {
+    if (contains(settings::surface_source_batch, simulation::current_batch) ||
+        simulation::current_batch == settings::n_batches) {
+      auto filename = settings::path_output + "surface_source." +
+                      std::to_string(simulation::current_batch);
+      // no batches specified for writing, write surface source bank
+      if (settings::surface_source_batch.size() == 0)
```

Comment:
In the case that the `surface_source_batch` vector is empty, we also want to make sure that we're only writing this file on the final batch right? Maybe a condition from above is misplaced?

Reply:
I think this is taken into account with the first condition. Only if the current batch is in the batch list or is the last batch will the file be written. In case it is, the condition you point out is to decide only the file name. However, doing the test I realized that if the last batch was not included in the list, it would still write it, so I made a change to fix that.